src/pybamm/citations.py

```diff
@@ -165,21 +154,13 @@ def _parse_citation(self, key):
 
                 # Add and register all citations
                 for key, entry in bib_data.entries.items():
-                    # Add to _all_citations dictionary
                     self._add_citation(key, entry)
-                    # Add to _papers_to_cite set
                     self._papers_to_cite.add(key)
-                    return
+                return
```

Comment:
This would skip the loop after the first item. That seemed like a bug, so I fixed it while doing the rest of the changes. If anyone thinks this was intended behavior, then I can switch it back or rewrite the block of code to be less confusing.